src/mococrw/distinguished_name.h

```diff
@@ -40,30 +41,25 @@ namespace mococrw
 class DistinguishedName
 {
 public:
-    /**
-     * Forward declaration of nested Builder class
-     *
-     * Definition is below.
-     *
-     */
     class Builder;
-
+    class CustomOrderBuilder;
     /**
      * Get the various attributes in this distinguished name.
      *
      * If an attribute has not been set, the corresponding method will return an
      * empty string.
      */
-    const std::string &commonName() const { return _commonName; }
-    const std::string &countryName() const { return _countryName; }
-    const std::string &localityName() const { return _localityName; }
-    const std::string &stateOrProvinceName() const { return _stateOrProvinceName; }
-    const std::string &organizationalUnitName() const { return _organizationalUnitName; }
-    const std::string &organizationName() const { return _organizationName; }
-    const std::string &pkcs9EmailAddress() const { return _pkcs9EmailAddress; }
-    const std::string &serialNumber() const { return _serialNumber; }
-    const std::string &givenName() const { return _givenName; }
-    const std::string &userId() const { return _userId; }
+    std::string commonName() const { return _getAttributeByNIDAsString(openssl::ASN1_NID::CommonName); }
```

Comment:
Since this changes the ABI of the program, we should probably increase the SOVERSION of the generated library. Currently, the SOVERSION handling is tied to the MoCOCrW version number (see `src/CMakeLists.txt:5`), which is not ideal.

I would propose to change the SOVERSION to `1.0` for this change, since the SOVERSION should only change when the ABI changes, and not when we release an ABI-compatible update with a newer version number.

Reply:
Sure

Reply:
These are currently two variables. Do you want to hard code 1.0 at line 5 in the cmake?

Reply:
Yes. It was a mistake to set the library SOVERSION to the version of the library. Those two are separate numbers.

Reply:
DONE. What about the mococrw version itself?

Reply:
I'd leave that as-is for now. We can do a commit to increase it later and push a tag on that commit.